.github/workflows/docker-builds.yaml

```
@@ -60,6 +63,20 @@ jobs:
             latest=false
             suffix=${{ matrix.flavor }}
 
+      - name: Generate tags for prefecthq/prefect
+        id: metadata-prod
+        uses: docker/metadata-action@v4
+        with:
+          images: prefecthq/prefect,enable=${{ github.event_name == 'release' }}
+          tags: |
+            type=raw,enable=${{ matrix.python-version == '3.10' }},value=2-latest
```

```suggestion
            type=raw,enable=${{ matrix.python-version == '3.10' && github.event.release.prerelease == 'false' }},value=2-latest
```
https://docs.github.com/developers/webhooks-and-events/webhooks/webhook-events-and-payloads?actionType=released#release

---

this relies on the maintainer ticking the 'this is a prerelease' button before publishing the release @madkinsz

---

That makes sense to me — there's also the `python{x}-latest` tags.

While it's on my mind, we'll eventually want the ability to release patches for old releases which means we'll need to check if the release is marked as the latest too 🤯 there's not an attribute for this so we'll probably need to make an API call for the latest release and see if the tags match. What a pain. I don't think that needs to happen now though.

---

@ddelange that's okay with me — seems like a pain to parse the tags.

---

> there's also the `python{x}-latest` tags.

did I fail to migrate properly? looks like missing here

---

> release patches for old releases

do you mean upstream patches? like `python:3.9` patched a security hole?

---

which reminds me we might want to set `pull: true` for exactly this reason, otherwise I guess it would eternally be using the base image as included in `buildcache`

---

> do you mean upstream patches? like python:3.9 patched a security hole?

I mean if we are on 2.7.7 but I want to release 2.6.10 with a backported fix.

---

But also yeah that other case is one we care about too! We may want to update the base image periodically for old releases.

---

> did I fail to migrate properly? looks like missing here

I guess it's `prefect:2-python3.x` i.e. https://github.com/PrefectHQ/prefect/blob/main/.github/workflows/release.yaml#L84

---

ok, this last one is included.

as for the other topic, I'm sure there is a not-too-ugly way of getting this info... maybe the build-push-action `latest` mechanic does these things out of the box 🤔

for now will apply the suggestions, feel free to unresolve of course 👍